.github/workflows/gh_pr_checks_approval.yml

```diff
@@ -5,73 +5,119 @@ on:
-            repo,
-            per_page,
-            event: 'pull_request',
-            status: 'action_required'
-          })).filter(workflow_run => workflow_run.head_sha === pull_request.head.sha);
-          if (workflow_runs.length === 0) {
-            return;
-          }
+  call_approval_checks_run:
+    needs: [ pull_request_opened, pull_request_approved ]
+    if: always() && contains(needs.*.result, 'success') && !(contains(needs.*.result, 'failure')) && contains(needs.*.outputs.run, 'true')
+    uses: ./.github/workflows/reuse_approve_checks_run.yml
+    with:
+      pull_request_number: ${{ fromJSON(needs.pull_request_opened.outputs.pull_request_number || needs.pull_request_approved.outputs.pull_request_number) }}
+      pull_request_head_sha: ${{ needs.pull_request_opened.outputs.pull_request_head_sha || needs.pull_request_approved.outputs.pull_request_head_sha }}
 
-          const [includeRiskFiles] = await github.paginate(pulls.listFiles, {
-            owner,
-            repo,
-            per_page,
-            pull_number: pull_request.number
-          }, ({ data: files }, done) => {
-            if (files.some(file => file.filename.startsWith('.github/workflows'))) {
-              done();
-              return [true];
-            }
-            return [];
-          });
+  checks_approval_comment:
+    needs: call_approval_checks_run
+    if: always() && needs.call_approval_checks_run.outputs.action_required == 'true'
+    runs-on: ubuntu-latest
+    env:
+      safety_changes_message: |
+        After a quick scan, I have approved workflow to run.
+      risky_changes_message: |
+        Sorry, due to risky changes, I can\'t approve workflow to run, our collaborators will handle it asap.
+      details_message: |
+        <details>
 
-          if (!includeRiskFiles) {
-            await Promise.all(workflow_runs.map(workflow_run => actions.approveWorkflowRun({
-              owner,
-              repo,
-              run_id: workflow_run.id,
-            })));
-          }
+        * SHA: ${{ needs.call_approval_checks_run.outputs.pull_request_head_sha }}
 
-          await issues.createComment({
-            owner,
-            repo,
-            issue_number: pull_request.number,
-            body: util.format(process.env[includeRiskFiles ? 'risky_changes_message' : 'safety_changes_message'], workflow_runs.length === 1 ? " the " : " ", workflow_runs.length > 1 ? "s" : ""),
-          });
+        </details>
+      tips_message: |
+        :label: **I would not automatically test new commits in this PR**, until our collaborators to verified. To become verified, they should either approve this PR or apply the `action(approve-checks)` label to this PR.
+        :label: **No need to worry about the status of `[gh] pull request merge guard / merge_guard (pull_request_target)` check**, once this PR is merged via `action(rebase-merge)` or `action(squash-merge)`, it will automatically trans to success status.
```

Comment:
it will automatically trans to success status
-> 
it will be automatically converted to successful status